controllers/operator/mongodbreplicaset_controller.go

```diff
@@ -244,6 +244,29 @@ func (r *ReconcileMongoDbReplicaSet) Reconcile(ctx context.Context, request reco
 		}
 	}
 
+	// Check if TLS is being disabled. If so, we need to lock replicas at the current member count
+	// to prevent scaling during the TLS disable operation. This decision is made once here and
+	// applied to both the StatefulSet and OM automation config.
+	tlsWillBeDisabled, err := checkIfTLSWillBeDisabled(conn, rs, log)
```

Comment:
what if we just block this with validation and say that it's not possible to change both member count and disabling TLS at the same time?